packages/evm/contracts/incentives/ERC20PeggedIncentive.sol

```diff
@@ -0,0 +1,148 @@
+        InitPayload memory init_ = abi.decode(data_, (InitPayload));
+        uint256 amount = init_.limit;
+
+        return abi.encode(
+            ABudget.Transfer({
+                assetType: ABudget.AssetType.ERC20,
+                asset: init_.asset,
+                target: address(this),
+                data: abi.encode(ABudget.FungiblePayload({amount: amount}))
+            })
+        );
+    }
+
+    /// @notice Claim the incentive with variable rewards
+    /// @param data_ The data payload for the incentive claim `(uint256signedAmount)`
+    /// @return True if the incentive was successfully claimed
+    function claim(address claimTarget, bytes calldata data_) external virtual override onlyOwner returns (bool) {
+        BoostClaimData memory boostClaimData = abi.decode(data_, (BoostClaimData));
+        uint256 signedAmount = abi.decode(boostClaimData.incentiveData, (uint256));
+
+        if (!_isClaimable(claimTarget, signedAmount)) revert NotClaimable();
+
+        if (totalClaimed + signedAmount > limit) revert ClaimFailed();
+
+        totalClaimed += signedAmount;
+        claims += 1;
+        asset.safeTransfer(claimTarget, signedAmount);
+
+        emit Claimed(claimTarget, abi.encodePacked(asset, claimTarget, signedAmount));
+        return true;
+    }
+
+    /// @inheritdoc AIncentive
+    function clawback(bytes calldata data_) external override onlyRoles(MANAGER_ROLE) returns (uint256, address) {
+        ClawbackPayload memory claim_ = abi.decode(data_, (ClawbackPayload));
+        (uint256 amount) = abi.decode(claim_.data, (uint256));
+
+        limit -= amount;
+
+        // Transfer the tokens back to the intended recipient
+        asset.safeTransfer(claim_.target, amount);
+        emit Claimed(claim_.target, abi.encodePacked(asset, claim_.target, amount));
+
+        return (amount, asset);
+    }
+
+    /// @notice Check if an incentive is claimable
+    /// @param claimTarget the address that could receive the claim
+    /// @return True if the incentive is claimable based on the data payload
+    /// @dev For the POOL strategy, the `bytes data` portion of the payload ignored
+    /// @dev The recipient must not have already claimed the incentive
+    function isClaimable(address claimTarget, bytes calldata data_) public view override returns (bool) {
+        BoostClaimData memory boostClaimData = abi.decode(data_, (BoostClaimData));
+        uint256 signedAmount = abi.decode(boostClaimData.incentiveData, (uint256));
+
+        return _isClaimable(claimTarget, signedAmount);
+    }
+
+    /// @notice Check if an incentive is claimable for a specific recipient
+    /// @param recipient_ The address of the recipient
+    /// @return True if the incentive is claimable for the recipient
+    function _isClaimable(address recipient_, uint256 amount) internal view returns (bool) {
+        return !claimed[recipient_] && (totalClaimed + amount) <= limit;
```

Comment:
is it necessary to enforce per-address single claims here in addition to the Validator level?

Reply:
Probably not but idk where-all `isClaimable` is being used at this point. I was mainly trying to maintain parity with [the existing ERC20Incentive](https://github.com/boostxyz/boost-protocol/blob/main/packages/evm/contracts/incentives/ERC20Incentive.sol#L143)